plugins/polio/api/dashboards/supply_chain.py

```diff
@@ -54,18 +58,71 @@ def get_stock_in_hand(self, obj, vaccine_stock):
 
         return self.context["stock_in_hand_cache"][cache_key]
 
-    def get_form_a_reception_date(self, vaccine_stock):
+    def get_form_a_reception_date(self, obj, vaccine_stock):
         latest_outgoing_stock_movement = (
-            OutgoingStockMovement.objects.filter(vaccine_stock=vaccine_stock).order_by("-form_a_reception_date").first()
+            OutgoingStockMovement.objects.filter(vaccine_stock=vaccine_stock, campaign=obj.campaign)
+            .order_by("-form_a_reception_date")
+            .first()
         )
         return latest_outgoing_stock_movement.form_a_reception_date if latest_outgoing_stock_movement else None
 
-    def get_destruction_report_reception_date(self, vaccine_stock):
-        latest_destruction_report = (
-            DestructionReport.objects.filter(vaccine_stock=vaccine_stock)
-            .order_by("-rrt_destruction_report_reception_date")
+    def get_destruction_report_reception_date(self, obj, vaccine_stock):
+        first_round_start_date = (
+            Round.objects.filter(campaign=obj.campaign)
+            .order_by("started_at")
+            .values_list("started_at", flat=True)
             .first()
         )
+
+        earliest_round_start_date = (
+            Round.objects.filter(campaign=OuterRef("pk")).order_by("started_at").values("started_at")[:1]
+        )
+
+        distinct_campaign_ids = (
+            Campaign.objects.filter(
+                country=vaccine_stock.country,
+                rounds__started_at__gt=first_round_start_date,
+                rounds__number=1,
+                account=self.context["request"].user.iaso_profile.account,
+            )
+            .order_by("id")
+            .distinct("id")
+            .values_list("id", flat=True)
+        )
+
+        campaigns_after_last_round = (
+            Campaign.objects.filter(id__in=Subquery(distinct_campaign_ids))
+            .annotate(earliest_round_start=Subquery(earliest_round_start_date))
+            .order_by("earliest_round_start")
+        )
+
+        next_campaign_start_date = None
+
+        for campaign in campaigns_after_last_round:
```

Comment:
would it not be better to order `campaigns_after_last_round` by descending order and break the loop once we have a value for `next_campaign_start_date`?

Also, since you already have the `Campaign` instance, you can probably prefetch the `rounds` i.o filtering on all `Round`s (dunno if it's really more performant)